public/scripts/slash-commands.js

```diff
@@ -1998,6 +1998,58 @@ export function initDefaultSlashCommands() {
             return '';
         },
     }));
+    SlashCommandParser.addCommandObject(SlashCommand.fromProps({
+        name: 'fnr',
+        aliases: ['find-and-replace','re','replace'],
+        callback: (async (args, text) => {
+            let mode = args.mode ?? 'literal';
+            if (args.find === '') {
+                toastr.error("Argument of '/fnr find=' cannot be empty!");
+                return '';
+            }
+            switch (mode) {
+                case 'regex':
+                    if (!(PARSER_MODE['STRICT_ESCAPING'])) {
+                        toastr.error("'/fnr mode=regex' may only be used with STRICT_ESCAPING enabled!");
+                        return '';
+                    }
+                    try {
+                        const [, pattern, flags] = args.find.match(/^\/(.+)\/([gimsuvy]*)$/);
+                        return text.replace(new RegExp(pattern, flags), args.replace ?? '');
+                    }
+                    catch {
+                        toastr.error(`${args.find} is not a valid RegEx pattern!`);
+                        return '';
+                    }
+                case 'literal':
+                    return text.replaceAll(args.find, args.replace ?? '');
+                default:
+                    toastr.error("Invalid '/fnr mode=' argument specified!");
+                    return '';
+            }
+        }),
+        returns: 'replacement',
+        namedArgumentList: [
+            SlashCommandNamedArgument.fromProps({
+                name: 'mode',
+                description: 'Replaces occurrence(s) of a pattern',
+                typeList: [ARGUMENT_TYPE.STRING],
+                defaultValue: 'literal',
+                enumList: ['literal', 'regex'],
+            }),
+            new SlashCommandNamedArgument(
+                'find', 'pattern to match against', [ARGUMENT_TYPE.STRING], true, false,
+            ),
+            new SlashCommandNamedArgument(
+                'replace', 'replacement text for matches', [ARGUMENT_TYPE.STRING], false, false,
```

Comment:
For a bit better clarity I would mark this as defaulting to empty string. Otherwise it could be mistaken that if you don't find a replace, it does nothing or something. Doesn't hurt to show what you actual replacer will be.
```suggestion
                'replace', 'replacement text for matches', [ARGUMENT_TYPE.STRING], false, false, '',
```